### ISSUE TITLE
Pre-commit hook passing locally, failing in CI

### ISSUE THREAD

**Issue by rpdelaney** [asottile/seed-isort-config#46]
Possibly related to  #7

You should be able to see the failed build [here](https://circleci.com/gh/rpdelaney/downforeveryone/39).

Here it is passing in my local environment:

```
$ git clone git@github.com:rpdelaney/downforeveryone.git
Cloning into 'downforeveryone'...
remote: Enumerating objects: 239, done.
remote: Counting objects: 100% (239/239), done.
remote: Compressing objects: 100% (138/138), done.
remote: Total 239 (delta 131), reused 185 (delta 78), pack-reused 0
Receiving objects: 100% (239/239), 48.37 KiB | 1.21 MiB/s, done.
Resolving deltas: 100% (131/131), done.
$ cd downforeveryone/
direnv: loading .envrc
direnv: export +POETRY_ACTIVE +VIRTUAL_ENV -PS2 ~PATH
$ pre-commit run --all
Fix End of Files.........................................................Passed
Mixed line ending........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for added large files..............................................Passed
Check that executables have shebangs.....................................Passed
Check for broken symlinks............................(no files to check)Skipped
Check JSON...........................................(no files to check)Skipped
Check for merge conflicts................................................Passed
Detect Private Key.......................................................Passed
Debug Statements (Python)................................................Passed
Check docstring is first.................................................Passed
Check Toml...............................................................Passed
Check Yaml...............................................................Passed
flake8...................................................................Passed
Shell Syntax Check.......................................................Passed
mypy.....................................................................Passed
seed isort known_third_party.............................................Passed
isort....................................................................Passed
black....................................................................Passed
$
```

Also, if I remove `known_third_party = []` from pyproject.toml then the hook creates an .isort.cfg like this:

```
[settings]
known_third_party = 
```

I haven't tried but I suspect that would also fail in CI?

**Comment by asottile** [comment]
iiiiinteresting, for me it fails locally as well:

```console
$ pre-commit run seed-isort-config --all-files
seed isort known_third_party.............................................Failed
hookid: seed-isort-config

Files were modified by this hook.

$ git diff
diff --git a/pyproject.toml b/pyproject.toml
index 1447c8b..2548352 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -40,7 +40,7 @@ target_version = ["py38"]
 combine_star = true
 indent='    '
 known_first_party = []
-known_third_party = []
+known_third_party = ["pytest", "requests"]
 length_sort = false
 line_length = 79
 multi_line_output = 3
```

Can you provide the following for me?

```bash
git status
ls -al
$(head -1 $(which pre-commit)
$(head -1 $(which pre-commit) | cut -d'!' -f2) --version --version
```

**Comment by rpdelaney** [comment]
```
$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
$ ls -al
total 72
drwxr-xr-x  9 ryan ryan   420 Dec 22 16:00 .
drwxrwxrwt 17 root root   580 Dec 22 16:00 ..
drwxr-xr-x  2 ryan ryan    60 Dec 22 15:18 bin
drwxr-xr-x  2 ryan ryan    60 Dec 22 15:18 .circleci
-rw-r--r--  1 ryan ryan    67 Dec 22 15:18 .coveragerc
drwxr-xr-x  2 ryan ryan   120 Dec 22 15:18 downforeveryone
-rw-r--r--  1 ryan ryan   420 Dec 22 15:18 .editorconfig
-rw-r--r--  1 ryan ryan   395 Dec 22 15:18 .envrc
drwxr-xr-x  8 ryan ryan   260 Dec 22 16:00 .git
-rw-r--r--  1 ryan ryan    13 Dec 22 15:18 .gitignore
-rw-r--r--  1 ryan ryan 11357 Dec 22 15:18 LICENSE
drwxr-xr-x  3 ryan ryan    60 Dec 22 15:18 .mypy_cache
-rw-r--r--  1 ryan ryan   593 Dec 22 15:18 mypy.ini
-rw-r--r--  1 ryan ryan 20454 Dec 22 15:18 poetry.lock
-rw-r--r--  1 ryan ryan  1258 Dec 22 15:18 .pre-commit-config.yaml
-rw-r--r--  1 ryan ryan  1130 Dec 22 16:00 pyproject.toml
-rw-r--r--  1 ryan ryan   107 Dec 22 15:18 pytest.ini
-rw-r--r--  1 ryan ryan  1927 Dec 22 15:18 README.rst
-rw-r--r--  1 ryan ryan   216 Dec 22 15:18 setup.cfg
drwxr-xr-x  2 ryan ryan   120 Dec 22 15:18 tests
drwxr-xr-x  5 ryan ryan   140 Dec 22 15:18 .venv
$ eval "$(command -v pre-commit)" --version --version
pre-commit 1.20.0
```

**Comment by asottile** [comment]
the last command was not to find pre-commit's version, but the python version that pre-commit is installed to

**Comment by asottile** [comment]
`env -i PYTHON` might also be useful, and `pre-commit run --verbose seed-isort-config` 🤔

**Comment by rpdelaney** [comment]
Ah

```
$ head -1 $(command -v pre-commit)
#!/home/ryan/.local/pipx/venvs/pre-commit/bin/python
$ /home/ryan/.local/pipx/venvs/pre-commit/bin/python --version --version
Python 3.8.0 (default, Oct 23 2019, 18:51:26)
[GCC 9.2.0]
$ env | grep -i python
$ pre-commit run --verbose seed-isort-config
[seed-isort-config] seed isort known_third_party.........................Passed
```

**Comment by asottile** [comment]
humm I wonder if it's `git` instead?

can you try

```
env | grep -i git
git ls-files -- '*.py'
```

the other thing to do would be to try and make a standalone virtualenv and see what `aspy.refactor_imports.classify.classify_import('pytest')` says -- it should be THIRD_PARTY but maybe there's something wacky happening here

I've never used `pipx` so that might be a difference as well, though from some reading it seems to do not-terribly-insane things so it should just work (?)

**Comment by rpdelaney** [comment]
I redacted a couple things

```
$ env | grep -i git
GIT_GLOB_PATHSPECS=0
GIT_NOGLOB_PATHSPECS=1
GIT_DISCOVER_ACROSS_FILESYSTEM=1
GIT_EDITOR=vim
$ git ls-files -- '*.py'
$ git ls-files
.circleci/config.yml
.coveragerc
.editorconfig
.envrc
.gitignore
.pre-commit-config.yaml
LICENSE
README.rst
bin/ship
downforeveryone/__init__.py
downforeveryone/cli.py
downforeveryone/isup.py
downforeveryone/useragents.py
mypy.ini
poetry.lock
pyproject.toml
pytest.ini
setup.cfg
tests/__init__.py
tests/test_cli.py
tests/test_isup.py
tests/test_useragents.py
```
Now that's kinda odd...

```
$ GIT_NOGLOB_PATHSPECS= GIT_GLOB_PATHSPECS= git ls-files -- '*.py'
downforeveryone/__init__.py
downforeveryone/cli.py
downforeveryone/isup.py
downforeveryone/useragents.py
tests/__init__.py
tests/test_cli.py
tests/test_isup.py
```
We have a winner!

**Comment by rpdelaney** [comment]
```
$ GIT_NOGLOB_PATHSPECS= GIT_GLOB_PATHSPECS= pre-commit run --all
Fix End of Files.........................................................Passed
Mixed line ending........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for added large files..............................................Passed
Check that executables have shebangs.....................................Passed
Check for broken symlinks............................(no files to check)Skipped
Check JSON...........................................(no files to check)Skipped
Check for merge conflicts................................................Passed
Detect Private Key.......................................................Passed
Debug Statements (Python)................................................Passed
Check docstring is first.................................................Passed
Check Toml...............................................................Passed
Check Yaml...............................................................Passed
flake8...................................................................Passed
Shell Syntax Check.......................................................Passed
mypy.....................................................................Passed
seed isort known_third_party.............................................Failed
hookid: seed-isort-config

Files were modified by this hook.

isort....................................................................Passed
black....................................................................Passed
```
Boom

**Comment by rpdelaney** [comment]
Looks like adding `GIT_NOGLOB_PATHSPECS` to `ENV_BLACKLIST` could fix this?

```diff
diff --git i/seed_isort_config.py w/seed_isort_config.py
index 66b51ef..910fd3c 100644
--- i/seed_isort_config.py
+++ w/seed_isort_config.py
@@ -13,7 +13,7 @@ from aspy.refactor_imports.classify import classify_import
 from aspy.refactor_imports.classify import ImportType


-ENV_BLACKLIST = frozenset(('GIT_LITERAL_PATHSPECS', 'GIT_GLOB_PATHSPECS'))
+ENV_BLACKLIST = frozenset(('GIT_LITERAL_PATHSPECS', 'GIT_GLOB_PATHSPECS', 'GIT_NOGLOB_PATHSPECS'))
 SUPPORTED_CONF_FILES = (
     '.editorconfig', '.isort.cfg', 'setup.cfg', 'tox.ini', 'pyproject.toml',
 )
```

**Comment by asottile** [comment]
hell yeah! nice

**Comment by asottile** [comment]
this has been released in v1.9.4 -- thanks again for the helpful debugging :)
